src/MvvmNano.Forms/MvvmNanoFormsPresenter.cs

```diff
@@ -90,21 +92,39 @@ public void SetApplication(Application application)
                 .ToArray();
         }
 
+        bool IsMasterDetailApplication()
+        {
+            return Application is MvvmNanoMasterDetailApplication;
+        }
+
+        public void ChangeRootViewModel<TViewModel>() where TViewModel : MvvmNanoViewModel
+        {
+            var viewName = GetViewNameByViewModel(typeof(TViewModel));
+            Type viewType = GetViewTypeByName(viewName); 
+            if(IsMasterDetailApplication() && viewType.GetTypeInfo().IsSubclassOf(typeof(NanoMasterDetailPage)))
+                ((MvvmNanoMasterDetailApplication)Application).SetUpMasterDetailPage<TViewModel>();    
```

Comment:
I am wondering if we need a `SetUpMasterDetailPage()` method in `MvvmNanoMasterDetailApplication`, or if it would be better to move this logic here to an overwritten `SetUpMainPage()`, so the presenter does not need to know anything about all that master detail stuff, it's just setting the main page as before.